modules/apps/29-fee/keeper/escrow.go

```diff
@@ -0,0 +1,136 @@
+		return sdkerrors.Wrapf(types.ErrRefundAccNotFound, "account with address: %s not found", refundAcc)
+	}
+
+	coins := identifiedFee.Fee.RecvFee
+	coins = coins.Add(identifiedFee.Fee.AckFee...)
+	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(
+		ctx, refundAcc, types.ModuleName, coins,
+	); err != nil {
+		return err
+	}
+
+	// Store fee in state for reference later
+	k.SetFeeInEscrow(ctx, identifiedFee)
+	return nil
+}
+
+// DistributePacketFees pays the acknowledgement fee & receive fee for a given packetId while refunding the timeout fee to the refund account associated with the Fee.
+func (k Keeper) DistributePacketFees(ctx sdk.Context, refundAcc, forwardRelayer string, reverseRelayer sdk.AccAddress, feeInEscrow types.IdentifiedPacketFee) {
+	// distribute fee for forward relaying
+	forward, err := sdk.AccAddressFromBech32(forwardRelayer)
+	if err == nil {
+		k.distributeFee(ctx, forward, feeInEscrow.Fee.RecvFee)
+	}
+
+	// distribute fee for reverse relaying
+	k.distributeFee(ctx, reverseRelayer, feeInEscrow.Fee.AckFee)
+
+	// refund timeout fee refund,
+	refundAddr, err := sdk.AccAddressFromBech32(refundAcc)
+	if err != nil {
+		panic(fmt.Sprintf("could not parse refundAcc %s to sdk.AccAddress", refundAcc))
+	}
+
+	// refund timeout fee for unused timeout
+	k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.TimeoutFee)
+
+	// removes the fee from the store as fee is now paid
+	k.DeleteFeeInEscrow(ctx, feeInEscrow.PacketId)
+}
+
+// DistributePacketsFeesTimeout pays the timeout fee for a given packetId while refunding the acknowledgement fee & receive fee to the refund account associated with the Fee
+func (k Keeper) DistributePacketFeesOnTimeout(ctx sdk.Context, refundAcc string, timeoutRelayer sdk.AccAddress, feeInEscrow types.IdentifiedPacketFee) {
+	// check if refundAcc address works
+	refundAddr, err := sdk.AccAddressFromBech32(refundAcc)
+	if err != nil {
+		panic(fmt.Sprintf("could not parse refundAcc %s to sdk.AccAddress", refundAcc))
+	}
+
+	// refund receive fee for unused forward relaying
+	k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.RecvFee)
+
+	// refund ack fee for unused reverse relaying
+	k.distributeFee(ctx, refundAddr, feeInEscrow.Fee.AckFee)
+
+	// distribute fee for timeout relaying
+	k.distributeFee(ctx, timeoutRelayer, feeInEscrow.Fee.TimeoutFee)
+
+	// removes the fee from the store as fee is now paid
+	k.DeleteFeeInEscrow(ctx, feeInEscrow.PacketId)
+}
+
+// distributeFee will attempt to distribute the escrowed fee to the receiver address.
+// If the distribution fails for any reason (such as the receiving address being blocked),
+// the state changes will be discarded.
+func (k Keeper) distributeFee(ctx sdk.Context, receiver sdk.AccAddress, fee sdk.Coins) {
+	// cache context before trying to send to reverse relayer
+	cacheCtx, writeFn := ctx.CacheContext()
+
+	err := k.bankKeeper.SendCoinsFromModuleToAccount(cacheCtx, types.ModuleName, receiver, fee)
```

Comment:
The call to `SendCoinsFromModuleToAccount` may fail if the [receiver address is blocked](https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/keeper/keeper.go#L302). And since the forward relayer address can be anything, there is a chance that this can be failed deliberately.

We may want to have some ways to handle the error, or at least have a log if the bank transfer fails. That way operators can at least found out that something is wrong is the fee module and begin investigation.